biochemist-python/chapters/Linear_Regression.ipynb

```diff
@@ -0,0 +1,752 @@
+    "`np.nan` is a NumPy function that results in a value of `NaN` (for not a number) as the output. This is typically used when there is no value in a cell, but it still preserves the datatype in that cell as a float. This is preferred to using a string such as 'Out of Range' which would change the datatype for those cells in the column to a string."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>Sample #</th>\n",
+       "      <th>A-595</th>\n",
+       "      <th>ProtConc</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>1</td>\n",
+       "      <td>0.599</td>\n",
+       "      <td>NaN</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>2</td>\n",
+       "      <td>0.682</td>\n",
+       "      <td>0.352738</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>3</td>\n",
+       "      <td>0.759</td>\n",
+       "      <td>0.519164</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>4</td>\n",
+       "      <td>1.013</td>\n",
+       "      <td>1.06816</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>5</td>\n",
+       "      <td>0.935</td>\n",
+       "      <td>0.899568</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>5</th>\n",
+       "      <td>6</td>\n",
+       "      <td>1.340</td>\n",
+       "      <td>NaN</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "   Sample #  A-595  ProtConc\n",
+       "0         1  0.599       NaN\n",
+       "1         2  0.682  0.352738\n",
+       "2         3  0.759  0.519164\n",
+       "3         4  1.013   1.06816\n",
+       "4         5  0.935  0.899568\n",
+       "5         6  1.340       NaN"
+      ]
+     },
+     "execution_count": 25,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "samples_df.loc[samples_df['A-595'] < 0.642, 'ProtConc'] = np.nan\n",
+    "samples_df.loc[samples_df['A-595'] > 1.188, 'ProtConc'] = np.nan\n",
+    "samples_df"
+   ]
```

Comment:
After showing this, it could also be an opportunity to introduce how to do this programmatically instead of using the hard-coded numbers of `0.642` and `1.188`. If you didn't want to look these up, you could use the `min` and `max` function on the data you used to fit the linear equation, ie

```
samples_df.loc[samples_df['A-595'] < results_df['A-595'].min(), 'ProtConc'] = np.nan
```

Reply:
I added a cell after this one and just before the exercise where I mentioned the .min() and .max() functions.